Lib/fontbakery/commands/check_profile.py

```diff
@@ -172,6 +177,13 @@ def parse_order(arg):
                       'collection against a selection of checks picked with `--checkid`.'
                       ''.format(', '.join(iterargs))
                       )
+
+  argument_parser.add_argument('-M','--multiprocessing', default=0, type=int,
+                      help=f'Use multi-processing to run the checks. The argument is the \n'
+                      f'number of worker processes. Use -1 to specify the number returned \n'
+                      f'by os.cpu_count()( = {os.cpu_count()}). Use 0 to run in single-processing mode \n'
```

Comment:
I'd rather say here "Use -1 to autodetect"

Reply:
Can do.

Reply:
I find `-1` counterintuitive. Why don't you have an optional argument that defaults to os.cpu_count()? That way on could just pass `-M` to enable multiprocessing. You could use `nargs="?"` with argparse, see:
https://docs.python.org/3/library/argparse.html#nargs

Another suggestion: you could name it `-j/--jobs` like GNU Make's, many people will be familiar with it already.

Reply:
I like Cosimo's suggestions ;-)

Reply:
I'm trying this out right now, but I'm rather unhappy about the handling of the optional numerical argument in the following case:

```
$ fontbakery check-googlefonts -j ../googleFonts/ofl/creepster/*.ttf
usage: fontbakery check-googlefonts [-h] [-c CHECKID] [-x EXCLUDE_CHECKID] [-v] [-l LOGLEVEL] [-m LOGLEVEL_MESSAGES] [--succinct] [-n] [-C] [-S] [-L] [--dark-theme] [--light-theme]
                                    [--json JSON_FILE] [--ghmarkdown MD_FILE] [--html HTML_FILE] [-g ITERATED_ARG] [-o ORDER] [-j [JOBS]]
                                    [fonts [fonts ...]]
fontbakery check-googlefonts: error: argument -j/--jobs: invalid int value: '../googleFonts/ofl/creepster/Creepster-Regular.ttf'
```

These examples work of course:

```
$ fontbakery check-googlefonts  ../googleFonts/ofl/creepster/*.ttf -j
$ fontbakery check-googlefonts -j -n ../googleFonts/ofl/creepster/*.ttf
```

It's just about the ambiguity that the optional `-j` argument followed by the `fonts` argument creates.

implementation:

```py
  argument_parser.add_argument('-j','--jobs', default=0, const=os.cpu_count(), type=int,
                      nargs='?', metavar='JOBS', dest='multiprocessing',
                      help='Use multi-processing to run the checks. The argument is the \n'
                      'number of worker processes. Without argument, the number returned \n'
                      'by os.cpu_count()( = %(const)s) is used. Use 0 to run in single-processing \n'
                      'mode (default %(default)s).'
                      )
```

I'd still choose this over the previous implementation, however, I think it's not ideal and one could argue that the improvement in intuitiveness is lost by the added ambiguity.

The name `-j/--jobs` will be kept for sure. 

We can have a small voting which version to use. React to this post with:

 * :tada:  for the "ambiguous" `nargs="?"` version 
 * :rocket: for the "counterintuitiv" `-1` version

Keep in mind, that multiprocessing may become the default and none of this would matter much.